Modified styles and assets

sanketbansal opened the pull request:
<!--
(Thanks for sending a pull request! .)
-->

<!--The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Introduce the first version of the collection editor." -->

<!-- Add issue number here. If you do not solve the issue entirely, please change the message e.g. "Addresses #IssueNumber -->


#### Changes proposed in this pull request:

<!-- Changes: Add here what changes were made in this issue and if possible provide links. -->
- Added assets from current EvalAI version.
- Added styles required for modifying pages.
- Added and modified the fonts.
- All added styles are palced in styles-port folder to support previous styles also.

<!-- Demo Link: Add here the link where you changes can be seen. -->
- Link to live demo: http://pr-154-evalai.surge.sh <!-- Replace XXX with your PR no: -->

<!-- Screenshots for the change: Add here the screenshot of the fix. -->

sanketbansal committed:
Fixed issue in font scss file

Removed '/' to provide proper assets path

sanketbansal committed:
Ported image assets

Ported the image assets from angularJs application into 'image-port' directory

sanketbansal committed:
Added fonts and font-awesome files from amgularjs

Fonts and font-awesome files are copied into the fonts folder

sanketbansal committed:
Modified fonts.scss file

Modified fonts.scss file to include fonts copied from angularjs application

sanketbansal commented:
@lunayach @Shekharrajak  I have made this seperate PR to reduce the file changes and repeated commits in other dependent PRs.

lunayach commented:
![Screenshot from 2019-06-23 23-23-49](https://user-images.githubusercontent.com/21227893/59978351-08a12d80-960e-11e9-8665-1fbfa6dbca9c.png)

@sanketbansal this is the screenshot that I took for the current UI. You may want to resolve some issues.

lunayach reviewed (changes requested):
@sanketbansal try to resolve the issues for this PR first so that we could review the other pending PR's.

sanketbansal commented:
@lunayach I have just added the styles in this PR. Not integrated them with the components that i will be doing in other PR for each component separately.

sanketbansal commented:
Also, I have kept the separate folder for these styles so that old styles can be applied.

lunayach commented:
Okay @sanketbansal, But could you refer all the pushed PR's which are depending on this (after attaching a screenshot in each of them), so that we could see if modified styles/assets are making sense.

sanketbansal commented:
#150  , #158 , #159 , #140 , #139 , #149 , #143 , #141 
All these PRs depends on these styles directly.

sanketbansal commented:
@lunayach  please review.

lunayach commented on src/styles/styles-port/modules/notification.scss:
```diff
@@ -0,0 +1,67 @@
+            text-decoration: none;
+        }
+    }
+    label {
```
Things like this could be removed.

lunayach commented:
@Shekharrajak @RishabhJain2018 Should we merge this PR first and we could see UI changes later in the above mentioned PR’s specifically?

Sanji515 commented:
@sanketbansal @lunayach @RishabhJain2018 @galipremsagar @Shekharrajak 

I think `vendor-port` css file is from the EvalAI project which is generated using gulp, which contains almost all the css code of EvalAI in one file and I think we should not copy exactly the same file.

lunayach commented:
Oh yes, it’s an auto-generated file. @sanketbansal please remove this. Thanks, @Sanji515 for spotting this.

Sanji515 commented:
And also I would suggest that as Angular 7 has differnet components and each components has it's `scss` file then we don't need to create these extra files and import them in `main.scss`. Even we don't need that `main.scss` file

lunayach reviewed (commented):
Ideally we should have component wise .scss. But since @sanketbansal has already done this way, we might need to discuss this.

sanketbansal commented:
@lunayach 

> Oh yes, it’s an auto-generated file. @sanketbansal please remove this. Thanks, @Sanji515 for spotting this.

Actually, vender-port.css files contain only the CSS related to third-party libraries i.e animate.css, angular material and toaster, trix and materialize.
So, I thought if we can use their raw CSS  instead of using these libraries as a whole. Then it will be great
Also, we can remove unused CSS from that. once complete implementing different components.

lunayach commented:
@sanketbansal Good thought, but I don't think we should ever make changes in the third-party libraries unless we have to. Otherwise, future developers/maintainers would have to scratch their heads for these trivial things :p.

And you need not worry about the unused stuff and all, Angular build when builds after optimisation, anyways makes files of very small sizes.

sanketbansal commented:
> @sanketbansal Good thought, but I don't think we should ever make changes in the third-party libraries unless we have to. Otherwise, future developers/maintainers would have to scratch their heads for these trivial things :p.

No, I am not making any changes in these libraries as I said I am just using their CSS properties as it is.
But can remove some unused properties if it is required.
Also if UI of the component lacks something then I add styles for the component specifically. Just like we do in angular 7.

Sanji515 commented:
I think the libraries they build, they use css for their things like rich text editor, angular material and in angular 7 if we want to use similar things then there are packages available. So instead of writing css we should directly install those packages along with their css.

sanketbansal commented:
yeah @Sanji515  we can do that, But I was talking about minor bugs such as margin or padding issue or color issue.

Sanji515 commented:
> But I was talking about minor bugs such as margin or padding issue or color issue.

@sanketbansal if you are talking to do these minor changes inside the CSS of those packages, then of course you can simply override them, let me know if I didn't get you

sanketbansal commented:
NO, I am talking about the changes in styles of component specifically inside **component.scss** file.

Shekharrajak reviewed (approved):
I believe you have made changes according to EvalAI properly.

Sanji515 commented:
@Shekharrajak But I think it will contain the two css files for one component one inside the component and another outside which is included inside the `main.scss` which is not good as structure wise and also we don't need to include the `vendor-port.css` file

lunayach commented:
@RishabhJain2018 just take a look at the comments thread above before merging.

RishabhJain2018 commented:
@sanketbansal Why are we adding `vendor-port.scss` file? I don't see it's need.

RishabhJain2018 commented:
I'm merging it for now to unblock you but we might need to clean this up in future.

sanketbansal commented:
> @sanketbansal Why are we adding `vendor-port.scss` file?

@RishabhJain2018 is explained that here  https://github.com/Cloud-CV/EvalAI-ngx/pull/154#issuecomment-505857371

RishabhJain2018 commented:
> So, I thought if we can use their raw CSS instead of using these libraries as a whole. 

Can you please list pros/cons of this?

RishabhJain2018 commented:
> And also I would suggest that as Angular 7 has differnet components and each components has it's scss file then we don't need to create these extra files and import them in main.scss. Even we don't need that main.scss file

Moreover, I'd prefer it to do this way which is a more cleaner approach.

sanketbansal commented:
I kept those styles separate in the **styles-port** folder so that these styles do not get mixed with the previous ones as @Sanji515  was also working on the project. 
I was looking to change them according to the preferred format once the important PRs get merged.

sanketbansal commented:
> > And also I would suggest that as Angular 7 has differnet components and each components has it's scss file then we don't need to create these extra files and import them in main.scss. Even we don't need that main.scss file
> 
> Moreover, I'd prefer it to do this way which is a more cleaner approach.

I have already moved the styles to their respective components. In other PRs.

lunayach commented:
> I have already moved the styles to their respective components. In other PRs.

@sanketbansal does this mean that you have moved the likes of `challenge-create.scss`, `challenge.scss` etc. to their respective component folders?

Shekharrajak commented:
Actually we were trying to unblock you but it seems if we merge this PR then it will remain same and it would be hard to refactor it. 

Every component must contain its own CSS but here we can see all things are in one folder (only common CSS should be outside the component folder).

Shekharrajak commented:
I think @Sanji515 can help to categorize the css and put it into the corresponding component folder. If there are multiple css files for one component then it should be in `compnent_name/css` (I just want to see good codebase structure).

sanketbansal commented:
@lunayach I have moved the challenge.scss and challenge-create file from the style-port folder into their respective components(#143, #150).

lunayach commented:
> @lunayach I have moved the challenge.scss and challenge-create file from the style-port folder into their respective components(#143, #150).

@sanketbansal  Okay, so what's wrong in doing the same in this PR itself? If not done, then again they would come under new changes in those PRs. That was the very purpose of this PR, I guess.

Shekharrajak commented:
Let's do one thing:

@Sanji515 can you clone this branch and try to put CSS files in the corresponding components and push new commits?

sanketbansal commented:
I have done some changes. Please review

lunayach commented:
@sanketbansal are the changes done?

sanketbansal commented:
@lunayach working on it. will commit those shortly.

sanketbansal commented:
@lunayach I have made those changes please review.
Also if any other changes required then please comment.

lunayach commented on src/app/components/challenge/challengeoverview/challengeoverview.component.spec.ts:
```diff
@@ -23,11 +25,54 @@ describe('ChallengeoverviewComponent', () => {
+    component.apiCall(updatedDescription);
+    expect(apiService.patchUrl).toHaveBeenCalled();
+    expect(globalService.showToast).toHaveBeenCalledWith('error', expectedErrorMsg);
+  });
```
Why is this changed in "modified styles.." PR?

sanketbansal replied:
will remove that

lunayach commented on src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html:
```diff
@@ -0,0 +1,135 @@
+      </div>
+    </div>
+  </div>
+</div>
```
Same for this.

sanketbansal replied:
I will remove that

lunayach commented on src/styles/main.scss:
```diff
@@ -0,0 +1,1590 @@
+@import "vendor-port.css";
```
How is this file different from `base.scss`? Doesn't `base.scss` serve a similar purpose?
@Sanji515 may also help in the discussion here.

lunayach commented on src/styles/main.scss:
```diff
@@ -0,0 +1,1590 @@
+
+
+/*ng cloak default styling*/
+
```
How is this file different from `base.scss`? Doesn't `base.scss` serve a similar purpose?
@Sanji515 may also help in the discussion here.

sanketbansal replied:
Yeah, so should I copy the styles from main.scss to base.scss?

lunayach replied:
Yes! just copy the relevant styles that are required now to the `base.scss`.

lunayach reviewed (changes requested):
1. Changes other than the 'modified styles and assets' are included.
2. Some other comments.

lunayach commented:
@sanketbansal Can you let us know which feature/dependency is (or going to be) dependent on `vendor-port.css`? Because it's an auto-generated file. (@Sanji515 may also join this discussion here.)

Sanji515 commented:
@lunayach @sanketbansal what I think that this PR even can be closed and the styles which are needed for all other PR's can be added in that only, cause the code I've written so far in the PR's is generalised in a way that we only need to write css if anything new there in the UI (for that we'll copy those css from this PR) otherwise we can use the already existing css which are in `base.scss`.

And also as this PR is not blocking to me and @sanketbansal for now so I think we can come to it later when all the current task are completed.

lunayach commented:
Yes, @Sanji515 I agree that we can add styles for the components on which we are working on. but then this PR (#140) has **137** file changes!  
Maybe for now we could include only the static files in this PR.
(@sanketbansal )

Sanji515 commented:
Then in `Modified homepage` PR, we can delete all the styles inside `styles-port`. Also we can delete the images which are included inside `images-port` as I see someone of them are already included inside `assets/images` in the master.

lunayach commented:
Yes! Does this sound good to you @sanketbansal?

sanketbansal commented:
> Then in `Modified homepage` PR, we can delete all the styles inside `styles-port`. Also we can delete the images which are included inside `images-port` as I see someone of them are already included inside `assets/images` in the master.

was not this the whole purpose from the beginning to merge this PR first to reduce the no of file changes and then remove unnecessary files in the corresponding PRs.

lunayach commented:
Yes, @sanketbansal that was the reason. If you could keep static files in this PR and resolve my other comments for the files in this PR, then we are good to go.

sanketbansal commented:
So I am removing all the styles from this PR. Okay

lunayach commented:
@sanketbansal  It would be better if you could reply inline to the comments raised in the Files as I had some doubts there as well.

sanketbansal commented:
> @sanketbansal Can you let us know which feature/dependency is (or going to be) dependent on `vendor-port.css`? Because it's an auto-generated file. (@Sanji515 may also join this discussion here.)

Auth pages use some styles from vendor-port also dashboard component use styles from vendor port. I am removing the vendor-port.scss anyway.

sanketbansal committed:
Modified styles

Added basic style and variable styles

sanketbansal commented:
@lunayach please review

lunayach commented on src/styles/base.scss:
```diff
@@ -1306,3 +1306,638 @@ td, th {
   display: block;
   color: #4d4d4d;
 }
+
```
I am not sure if all these styles are (or are going to be) used. In future PR's, @sanketbansal @Sanji515 try to remove redundant and unused styles from here.

sanketbansal replied:
yeah, I think that we can do in other dependent PRs

Sanji515 replied:
@lunayach yeah of course some changes are new but most of them are already on the master, this is the reason I'm saying to close this PR and do the css changes in their respective PR so that it would be easy to track what should not be included in the PR. These fonts can also be included in their respective PR's

lunayach replied:
I don't think that the `base.scss` is having changes for the already pushed PR's. That should not be the case anyways. Is it correct, @sanketbansal?

sanketbansal replied:
so, @Sanji515 can you tell me the changes which have already been made in the master?
Then I will remove them

sanketbansal replied:
@lunayach I have tried to added changes related to my PRs.
Also, U and @Sanji515  can run the PR on local machine and check if its hindering with already made changes.

lunayach replied:
Ah, let's not go this way then. Use updated master's `base.scss` for all your future PR's. 
Let's also remove `base.scss` from this PR. (That is, only static files for now.)
That way, I guess it would be easy to keep a track of all the changes done for a particular functionality.

sanketbansal replied:
I have checked there is no duplicate styles present in base.scss file. Also no style is hindering with the master branch. Also these added styles are required by the other components.

lunayach replied:
(IMO) if `base.scss` is also having styles for already pushed PR's, then we should not include it in this PR. We can easily afford one extra file change in the form of `base.scss` for future PR's.

Also, can you confirm for @Sanji515's comment:

> @sanketbansal The organizations' images that you put inside the organizations' folder are already there inside images/partners folder. So I think you should check for the duplicate images there and at other places also.

So, that means only static file changes for this PR. @sanketbansal I understand that you had to make lot's of changes for this (and other related) PR's, but to keep the project structure clean, I guess, it's worth it.

sanketbansal replied:
That's why I was keeping a separate file of my changes. And was looking to add that into base.scss after all PRs got merged.

lunayach replied:
But can we add those changes PR by PR as we proceed (in `base.scss`)?

sanketbansal replied:
Any issue with variable.scss file?
Also, should I make changes in base.scss in other PRs or not?

lunayach replied:
`Variable.scss` is fine.
Sure, you can make changes in `base.scss` for the functionality that the other PR introduces.

sanketbansal replied:
> But can we add those changes PR by PR as we proceed (in `base.scss`)?

For that I have to check each and every style of the HTML in other PR and then have to search whether that is already present or not which is going to be a tedious task.

lunayach replied:
I think to avoid possible duplicities and adding those things only which we actually need, it should be done. We could also take @RishabhJain2018's take on this.

RishabhJain2018 replied:
I agree with @lunayach.

lunayach reviewed (approved):
@RishabhJain2018 I think we can merge this now.

Sanji515 commented:
@sanketbansal Also I would suggest to remove all the images from `image-port` as you have ported all the images of current version in which some of them are not being used and also it contains some images which are already on the `master`

sanketbansal committed:
Removed duplicate styles

Duplicate styles remode from base.scss file

lunayach reviewed (changes requested):
@sanketbansal also move the images from `images-port` to `images` folder.

sanketbansal committed:
Removed duplicate assets

Duplicate assets removed from image-port folder

lunayach committed:
Merge branch 'master' into modified-style-assets

lunayach commented:
@sanketbansal I still see the `images-port` folder. Did you move all the images?

sanketbansal committed:
Removed image-port folder

Moved assets into image-port folder

Sanji515 commented:
@sanketbansal The organizations images that you put inside the `organizations` folder are already there inside `images/partners` folder. So I think you should check for the duplicate images there and at other places also.

sanketbansal commented:
There were only 7 duplicate image assets present which I have removed.

> @sanketbansal The organizations images that you put inside the `organizations` folder are already there inside `images/partners` folder. So I think you should check for the duplicate images there and at other places also.

yeah, I know that I am using organization folder in my PRs and did not removed partner folder so that in case you will be using that.
Anyways, I have removed that organization folder.

sanketbansal committed:
Removed common styles

Common styles removed from styles folder

sanketbansal commented:
removed styles from base.scss. please check

lunayach reviewed (approved):
@RishabhJain2018 this seems good to me now. This will not overlap/duplicate the already pushed (or future) changes.

sanketbansal commented:
@RishabhJain2018 I am waiting for this PR to be merged

sanketbansal commented:
@lunayach is it okay if I add a common.scss file to include styles which is common for different components.

lunayach commented:
@sanketbansal if the styles are common, they'll anyways be reflected for only the first PR that'll we merge. I don't think we need another `common.scss`.

sanketbansal commented:
> @sanketbansal if the styles are common, they'll anyways be reflected for only the first PR that'll we merge. I don't think we need another `common.scss`.

okay
